salt/utils/minions.py

```diff
@@ -224,7 +224,7 @@ def _check_list_minions(self, expr, greedy):  # pylint: disable=unused-argument
             expr = [m for m in expr.split(',') if m]
         minions = self._pki_minions()
         return {'minions': [x for x in expr if x in minions],
-                'missing': [x for x in expr if x not in minions]}
```

Comment:
This is not ok.

This was a specific feature added, and would rip out the behavior that shows regular list minions aren't specified.

`L@MINION_NOT_SPECIFIED` would not return that minion_not_specified is not a minion anymore.

Reply:
That is odd. With an unmodified version of Salt (2018.3.0-26.1 from SUSE) I see this:
```
salt -C 'L@foo' grains.get os
No minions matched the target. No command was sent, no jid was assigned.
ERROR: No return received
```
So you'd expect to get a return from the missing minion here or did I misunderstand you? Thanks!

Reply:
See [new comment](https://github.com/saltstack/salt/pull/49435#issuecomment-417671177) please.